sway_libs/src/signed_integers/i128.sw

```diff
@@ -0,0 +1,173 @@
+
+    /// The size of this type in bits.
+    pub fn bits() -> u32 {
+        128
+    }
+
+    /// Helper function to get a negative value of unsigned number
+    pub fn neg_from(value: U128) -> Self {
+        Self {
+            underlying: ~Self::indent() - value,
+        }
+    }
+
+    /// Helper function to get a positive value from unsigned number
+    fn from_uint(value: U128) -> Self {
+        // as the minimal value of I128 is -~I128::indent() (1 << 63) we should add ~I128::indent() (1 << 63) 
+        let underlying: U128 = value + ~Self::indent();
+        Self { underlying }
+    }
+}
+
+impl core::ops::Add for I128 {
+    /// Add a I128 to a I128. Panics on overflow.
+    fn add(self, other: Self) -> Self {
+        // subtract 1 << 63 to avoid double move
+        ~Self::from(self.underlying - ~Self::indent() + other.underlying)
+    }
+}
+
+impl core::ops::Subtract for I128 {
+    /// Subtract a I128 from a I128. Panics of overflow.
+    fn subtract(self, other: Self) -> Self {
+        let mut res = ~Self::new();
+        if self > other {
+            // add 1 << 63 to avoid loosing the move
+            res = ~Self::from(self.underlying - other.underlying + ~Self::indent());
+        } else {
+            // subtract from 1 << 63 as we are getting a negative value
+            res = ~Self::from(~Self::indent() - (other.underlying - self.underlying));
+        }
+        res
+    }
+}
+
+impl core::ops::Multiply for I128 {
+    /// Multiply a I128 with a I128. Panics of overflow.
+    fn multiply(self, other: Self) -> Self {
+        let mut res = ~Self::new();
+        if (self.underlying > ~Self::indent()
+            || self.underlying == ~Self::indent())
+            && (other.underlying > ~Self::indent()
+            || other.underlying == ~Self::indent())
+        {
+            res = ~Self::from((self.underlying - ~Self::indent()) * (other.underlying - ~Self::indent()) + ~Self::indent());
+        } else if self.underlying < ~Self::indent()
+            && other.underlying < ~Self::indent()
+        {
+            res = ~Self::from((~Self::indent() - self.underlying) * (~Self::indent() - other.underlying) + ~Self::indent());
+        } else if (self.underlying > ~Self::indent()
+            || self.underlying == ~Self::indent())
+            && other.underlying < ~Self::indent()
+        {
+            res = ~Self::from(~Self::indent() - (self.underlying - ~Self::indent()) * (~Self::indent() - other.underlying));
+        } else if self.underlying < ~Self::indent()
+            && (other.underlying > ~Self::indent()
+            || other.underlying == ~Self::indent())
+        {
+            res = ~Self::from(~Self::indent() - (other.underlying - ~Self::indent()) * (~Self::indent() - self.underlying));
+        }
+        res
+    }
+}
+
+impl core::ops::Divide for I128 {
+    /// Divide a I128 by a I128. Panics if divisor is zero.
+    fn divide(self, divisor: Self) -> Self {
+        assert(divisor != ~Self::new());
```

Comment:
This should be a `require` statement instead of an `assert` and return an error. `assert` should be reserved for testing.

Reply:
Resolved https://github.com/FuelLabs/sway-libs/pull/30/commits/ac6893f12c30e238f0df4a7a16da28e17daf40ad